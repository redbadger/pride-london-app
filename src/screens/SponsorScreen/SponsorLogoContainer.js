// @flow
import React from "react";
import { View, StyleSheet, Image, Linking } from "react-native";
import type { ViewStyleProp } from "react-native/Libraries/StyleSheet/StyleSheet";
import SponsorLogo from "./SponsorLogo";
import sponsorHeadline from "../../../assets/images/sponsorHeadline.png";
import sponsorGold from "../../../assets/images/sponsorGold.png";
import sponsorSilver from "../../../assets/images/sponsorSilver.png";
import sponsorBronze from "../../../assets/images/sponsorBronze.png";
import Text from "../../components/Text";
import Touchable from "../../components/Touchable";
import text from "../../constants/text";
import { sponsorLogoBackgroundColor } from "../../constants/colors";
import type { SponsorLevel, Sponsor } from "../../data/sponsor";
import type { LocalizedFieldRef } from "../../data/localized-field-ref";

import locale from "../../data/locale";

type Props = {
  sponsorLevel: SponsorLevel,
  sponsors: Sponsor[],
  getAssetUrl: LocalizedFieldRef => string,
  getAssetSize: LocalizedFieldRef => { width: number, height: number },
  style?: ViewStyleProp
};

const sponsorLevelIcons = {
  Headline: sponsorHeadline,
  Gold: sponsorGold,
  Silver: sponsorSilver,
  Bronze: sponsorBronze
};

const SponsorLogoContainer = ({
  sponsorLevel,
  getAssetUrl,
  getAssetSize,
  sponsors,
  style
}: Props) => (
  <View style={style}>
    <View style={styles.header}>
      <Image source={sponsorLevelIcons[sponsorLevel]} />
      <Text style={styles.sponsorHeading} type="h3" color="lightNavyBlueColor">
        {sponsorLevel} {text.sponsorLevelHeading}
      </Text>
    </View>

    <View style={styles.logoRow}>
      {sponsors &&
        sponsors.map((sponsor, index) => (
          <View
            key={sponsor.sys.id}
            style={
              sponsorLevel === "Headline" || sponsorLevel === "Gold"
                ? [
                    styles.tileTwoColumns,
                    index % 2 !== 0 && styles.optionalePadding
                  ]
                : [
                    styles.tileThreeColumns,
                    index % 3 !== 0 && styles.optionalePadding
                  ]
            }
          >
            <Touchable
              style={styles.sponsorTile}
              onPress={() => Linking.openURL(sponsor.fields.sponsorUrl[locale])}
              accessibilityTraits={["link"]}
              testID={`sponsor-tile-${sponsor.sys.id}`}
            >
              <SponsorLogo
                sponsorName={sponsor.fields.sponsorName[locale]}
                sponsorLevel={sponsor.fields.sponsorLevel[locale]}
                sponsorLogoUrl={getAssetUrl(sponsor.fields.sponsorLogo)}
                sponsorLogoSize={getAssetSize(sponsor.fields.sponsorLogo)}
              />
            </Touchable>
          </View>
        ))}
    </View>
  </View>
);

SponsorLogoContainer.defaultProps = {
  style: {}
};

const styles = StyleSheet.create({
  header: {
    alignItems: "center",
    flexDirection: "row"
  },
  tileTwoColumns: {
    width: "50%",
    marginBottom: 8
  },
  tileThreeColumns: {
    width: "33%",
    marginBottom: 8
  },
  optionalePadding: {
    paddingLeft: 8
  },
  logoRow: {
    flexDirection: "row",
    flexWrap: "wrap",
    marginTop: 12
  },
  sponsorTile: {
    height: 100,
    alignItems: "center",
    backgroundColor: sponsorLogoBackgroundColor
  },
  sponsorHeading: {
    marginLeft: 5,
    paddingTop: 3
  }
});

export default SponsorLogoContainer;
