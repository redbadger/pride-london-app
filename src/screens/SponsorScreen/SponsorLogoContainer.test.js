// @flow
import React from "react";
import { Linking } from "react-native";
import { shallow } from "enzyme";
import SponsorLogoContainer from "./SponsorLogoContainer";
import type { Sponsor } from "../../data/sponsor";

let openURLSpy;
beforeEach(() => {
  openURLSpy = jest
    .spyOn(Linking, "openURL")
    .mockImplementation(() => Promise.resolve());
});

const generateSponsors = (count = 2): Sponsor[] =>
  Array.from(Array(count)).map(
    (_, i) =>
      ({
        contentType: "sponsor",
        locale: "en-GB",
        id: String(i + 1),
        fields: {
          sponsorName: "some other",
          sponsorLogo: {
            sys: { id: `asset${i + 1}` }
          },
          sponsorUrl: "http://example.com",
          sponsorLevel: "Gold"
        },
        revision: 1
      }: Sponsor)
  );

const getAssetSource = jest.fn().mockReturnValue({
  uri: "http://example.com/image.png",
  width: 1,
  height: 1
});

describe("SponsorLogoContainer Component", () => {
  it("renders correctly with Gold Level", () => {
    const output = shallow(
      <SponsorLogoContainer
        sponsorLevel="Gold"
        sponsors={generateSponsors(2)}
        getAssetSource={getAssetSource}
      />
    );
    expect(output).toMatchSnapshot();
  });

  it("renders correctly with Silver Level", () => {
    const output = shallow(
      <SponsorLogoContainer
        sponsorLevel="Silver"
        sponsors={generateSponsors(2)}
        getAssetSource={getAssetSource}
      />
    );
    expect(output).toMatchSnapshot();
  });

  it("navigates to sponsor link onPress", () => {
    const output = shallow(
      <SponsorLogoContainer
        sponsorLevel="Silver"
        sponsors={generateSponsors(2)}
        getAssetSource={getAssetSource}
      />
    );
    const sponsorLink = output.find({ testID: "sponsor-tile-1" });
    sponsorLink.simulate("press");

    expect(openURLSpy).toHaveBeenCalledWith("http://example.com");
  });
});

afterEach(() => {
  openURLSpy.mockRestore();
});
