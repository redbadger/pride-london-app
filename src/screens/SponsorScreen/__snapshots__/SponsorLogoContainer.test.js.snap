// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SponsorLogoContainer Component renders correctly with Gold Level 1`] = `
<Component
  style={Object {}}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/sponsorGold.png",
        }
      }
    />
    <Text
      color="lightNavyBlueColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "marginLeft": 5,
          "paddingTop": 3,
        }
      }
      type="h3"
    >
      Gold
       
      partners
    </Text>
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": 12,
      }
    }
  >
    <Component
      key="1"
      style={
        Array [
          Object {
            "marginBottom": 8,
            "width": "50%",
          },
          false,
        ]
      }
    >
      <Touchable
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "link",
          ]
        }
        delayPressIn={50}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(243, 243, 243, 0.5)",
            "height": 100,
          }
        }
        testID="sponsor-tile-1"
      >
        <SponsorLogo
          sponsorLevel="Gold"
          sponsorLogo={
            Object {
              "sys": Object {
                "id": "asset1",
              },
            }
          }
          sponsorName="some other"
        />
      </Touchable>
    </Component>
    <Component
      key="2"
      style={
        Array [
          Object {
            "marginBottom": 8,
            "width": "50%",
          },
          Object {
            "paddingLeft": 8,
          },
        ]
      }
    >
      <Touchable
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "link",
          ]
        }
        delayPressIn={50}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(243, 243, 243, 0.5)",
            "height": 100,
          }
        }
        testID="sponsor-tile-2"
      >
        <SponsorLogo
          sponsorLevel="Gold"
          sponsorLogo={
            Object {
              "sys": Object {
                "id": "asset2",
              },
            }
          }
          sponsorName="some other"
        />
      </Touchable>
    </Component>
  </Component>
</Component>
`;

exports[`SponsorLogoContainer Component renders correctly with Silver Level 1`] = `
<Component
  style={Object {}}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/sponsorSilver.png",
        }
      }
    />
    <Text
      color="lightNavyBlueColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "marginLeft": 5,
          "paddingTop": 3,
        }
      }
      type="h3"
    >
      Silver
       
      partners
    </Text>
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": 12,
      }
    }
  >
    <Component
      key="1"
      style={
        Array [
          Object {
            "marginBottom": 8,
            "width": "33%",
          },
          false,
        ]
      }
    >
      <Touchable
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "link",
          ]
        }
        delayPressIn={50}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(243, 243, 243, 0.5)",
            "height": 100,
          }
        }
        testID="sponsor-tile-1"
      >
        <SponsorLogo
          sponsorLevel="Gold"
          sponsorLogo={
            Object {
              "sys": Object {
                "id": "asset1",
              },
            }
          }
          sponsorName="some other"
        />
      </Touchable>
    </Component>
    <Component
      key="2"
      style={
        Array [
          Object {
            "marginBottom": 8,
            "width": "33%",
          },
          Object {
            "paddingLeft": 8,
          },
        ]
      }
    >
      <Touchable
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "link",
          ]
        }
        delayPressIn={50}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(243, 243, 243, 0.5)",
            "height": 100,
          }
        }
        testID="sponsor-tile-2"
      >
        <SponsorLogo
          sponsorLevel="Gold"
          sponsorLogo={
            Object {
              "sys": Object {
                "id": "asset2",
              },
            }
          }
          sponsorName="some other"
        />
      </Touchable>
    </Component>
  </Component>
</Component>
`;
